Ginger/GingerCoreNET/Run/Context.cs

```diff
@@ -112,6 +131,144 @@ public TargetBase Target
             }
         }
 
+        private string mTargetApplication;
+        public string TargetApplication
+        {
+            get
+            {
+                return mTargetApplication;
+            }
+            set
+            {
+                if (mTargetApplication != value)
+                {
+                    mTargetApplication = value;
+                    OnPropertyChanged(nameof(TargetApplication));
+                }
+            }
+        }
+
+        private ApplicationAgent mAppAgent;
+        public ApplicationAgent AppAgent
+        {
+            get
+            {
+                return mAppAgent;
+            }
+            set
+            {
+                if (mAppAgent != value)
+                {
+                    mAppAgent = value;
+                    mAppAgent.PropertyChanged += AppAgent_PropertyChanged;
+                    Agent = mAppAgent.Agent;
+                    OnPropertyChanged(nameof(AppAgent));
+                }
+            }
+        }
+        
+        private Agent mAgent;
+        public Agent Agent
+        {
+            get
+            {
+                return mAgent;
+            }
+            set
+            {
+                if (mAgent != value)
+                {
+                    mAgent = value;
+                    mAgent.PropertyChanged += Agent_PropertyChanged;
+                    OnPropertyChanged(nameof(Agent));
+                }
+            }
+        }
+        
+        private string mAgentStatus;
+        public string AgentStatus
+        {
+            get
+            {
+                return mAgentStatus;
+            }
+            set
+            {
+                if (mAgentStatus != value)
+                {
+                    mAgentStatus = value;
+                    OnPropertyChanged(nameof(AgentStatus));
+                }
+            }
+        }
+
+        private ePlatformType mActivityPlatform;
+        public ePlatformType ActivityPlatform
+        {
+            get
+            {
+                return mActivityPlatform;
+            }
+            set
+            {
+                if (mActivityPlatform != value)
+                {
+                    mActivityPlatform = value;
+                    OnPropertyChanged(nameof(ActivityPlatform));
+                }
+            }
+        }
+
+        /// <summary>
+        /// This event is used to handle the Activity's TargetApplciation changed functionality
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void Activity_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
```

Comment:
move all PropertyChanged and SetAgent() to Automate page

Reply:
Moved to Automate Page